tests/unit/v1/test__helpers.py

```diff
@@ -20,6 +20,11 @@
 import mock
 
 
+class AsyncMock(mock.MagicMock):
```

Comment:
Why do you need to define your own AsyncMock rather than using https://docs.python.org/3/library/unittest.mock.html#unittest.mock.AsyncMock ?

Reply:
I didn't know that existed actually. Previously I was using AsyncMock provided by the [`asyncmock` package](https://pypi.org/project/asyncmock/), but it didn't work properly with some of the unittest features. I'll take a look at using this.

Reply:
This library function is only available for Python 3.8 onwards, so unfortunately we can't use it. Is the current solution acceptable in this case?

Reply:
Yep sounds good, thanks for looking into it.